api/validation.go

```diff
@@ -74,11 +84,48 @@ func (api *API) validateAndHydrateDimensions(v dataset.Version, dims []model.Dim
 	}
 }
 
+// Used for POST Dimension when the filter type is multivariate.
+// Extra logic required to hydrate dimension using the default categorisation of the dimension.
+func (api *API) HydrateMultivariateDimensions(dimensions []model.Dimension, pType string) ([]model.Dimension, error) {
+	ctx := context.Background()
+	hydratedDimensions := make([]model.Dimension, 0)
+	var finalLabel string
+
+	for _, dim := range dimensions {
+		finalDimension := dim.Name
+		node, err := api.getCantabularDimension(ctx, pType, dim.Name)
+		if err != nil {
+			return nil, errors.Wrap(err, "error in cantabular response")
+		}
+
+		finalDimension, finalLabel, err = api.CheckDefaultCategorisation(node.Name, pType)
+		if err != nil {
+			return nil, err
+		}
+
+		if dim.Options == nil {
+			dim.Options = []string{}
+		}
+
+		hydratedDimensions = append(hydratedDimensions, model.Dimension{
+			Name:                  finalDimension,
+			ID:                    finalDimension,
+			Label:                 finalLabel,
+			DefaultCategorisation: finalDimension,
+			Options:               dim.Options,
+			IsAreaType:            dim.IsAreaType,
+			FilterByParent:        dim.FilterByParent,
+		})
+
+	}
+	return hydratedDimensions, nil
+}
+
 // getCantabularDimensions pulls full dimension information from Cantabular using the names of the provided
 // dimensions.
 // NOTE: when we hydrate the dimensions, we will be using the name as the id, and filling out the dimensions
 // using the same value for both.
-func (api *API) getCantabularDimensions(ctx context.Context, dimensions []model.Dimension, pType string) ([]model.Dimension, error) {
+func (api *API) getCantabularDimensions(ctx context.Context, dimensions []model.Dimension, pType string, postDimension bool) ([]model.Dimension, error) {
```

Comment:
Do we need postDimension ?

Reply:
yeah we do need it as the changes in this ticket are just for POST dimension, but also concern POST filter in multivariate cases

Reply:
But where is is input param being used in that function?